bin/extract_doc_versions.pl

```diff
@@ -144,20 +147,20 @@ ()
   print $outfile "<tr><th>language</th><th>Sum up to date</th><th>Sum translated</th></tr>\n";
 
   # number of english files to translate
-  my $sum_files=scalar keys %{$svninfo{"en"}};
+  my $sum_files=scalar keys %{$gitinfo{"en"}};
 
   # loop through languages
-  foreach my $lang (sort keys %svninfo) {
+  foreach my $lang (sort keys %gitinfo) {
     # loop through filenames
     my $up_to_date=0;
-    foreach my $dir_file (keys %{$svninfo{"en"}}) {
-      if (exists $svninfo{$lang}{$dir_file}) {
-        if ($svninfo{$lang}{$dir_file}{'version'} >= $svninfo{"en"}{$dir_file}{'version'}) {
+    foreach my $dir_file (keys %{$gitinfo{"en"}}) {
+      if (exists $gitinfo{$lang}{$dir_file}) {
+        if ($gitinfo{$lang}{$dir_file}{'version'} >= $gitinfo{"en"}{$dir_file}{'version'}) {
```

Comment:
Hi Angelos,
Yes, I did work out that git version is a hash, so 
($gitinfo{$lang}{$dir_file}{'version'} actually points to a timestamp.

On 21/02/2016 10:43 am, Angelos Tzotsos wrote:
>
> In bin/extract_doc_versions.pl 
> <https://github.com/OSGeo/OSGeoLive-doc/pull/73#discussion_r53559343>:
>
> >      # loop through filenames
> >      my $up_to_date=0;
> > -    foreach my $dir_file (keys %{$svninfo{"en"}}) {
> > -      if (exists $svninfo{$lang}{$dir_file}) {
> > -        if ($svninfo{$lang}{$dir_file}{'version'} >= $svninfo{"en"}{$dir_file}{'version'}) {
> > +    foreach my $dir_file (keys %{$gitinfo{"en"}}) {
> > +      if (exists $gitinfo{$lang}{$dir_file}) {
> > +        if ($gitinfo{$lang}{$dir_file}{'version'} >= $gitinfo{"en"}{$dir_file}{'version'}) {
>
> The git version is a hash and there is no guarantee that the 
> comparison will work.
> The datetimes of the commits should be used instead.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/OSGeo/OSGeoLive-doc/pull/73/files#r53559343>.
>

-- 
Cameron Shorter,
Software and Data Solutions Manager
LISAsoft
Suite 112, Jones Bay Wharf,
26 - 32 Pirrama Rd, Pyrmont NSW 2009

P +61 2 9009 5000,  W www.lisasoft.com,  F +61 2 9009 5099